pkg/session_recording/recordings.jsx

```diff
@@ -340,6 +340,77 @@
     let RecordingList = class extends React.Component {
         constructor(props) {
             super(props);
+            this.sortingBy = this.sortingBy.bind(this);
+            this.sortWay = this.sortWay.bind(this);
+            this.sorting = this.sorting.bind(this);
+            this.sortByUser = this.sortByUser.bind(this);
+            this.sortBySince = this.sortBySince.bind(this);
+            this.sortByUntil = this.sortByUntil.bind(this);
+            this.sortByDuration = this.sortByDuration.bind(this);
+            this.calcDuration = this.calcDuration.bind(this);
+            this.drawSortArrow = this.drawSortArrow.bind(this);
+            this.state = {
+                sorting_field: "start",
+                sorting_way: "desc",
+            };
+        }
+
+        drawSortArrow(elem) {
+            $('#sort_arrow').remove();
+            let type = this.state.sorting_way ? "asc" : "desc";
+            let arrow = '<i id="sort_arrow" class="fa fa-sort-' + type + '" aria-hidden="true"></i>';
+            $(this.refs[elem]).append(arrow);
+        }
+
+        sortWay() {
+            if (this.state.sorting_way) {
+                this.setState({sorting_way: false});
+            } else {
+                this.setState({sorting_way: true});
+            }
+        }
+
+        sortingBy(field) {
+            this.setState({sorting_field: field});
+            this.sortWay();
+        }
+
+        sortByUser() {
+            this.sortingBy('user');
+        }
+
+        sortBySince() {
+            this.sortingBy('start');
+        }
+
+        sortByUntil() {
+            this.sortingBy('end');
+        }
+
+        sortByDuration() {
+            this.sortingBy('duration');
+        }
+
+        sorting(list) {
+            let field = this.state.sorting_field;
+            let sort_way = this.state.sorting_way;
+
+            if (sort_way === true) {
```

Comment:
If you worry about the value types screwing up, perhaps write a validation function for the whole object and assert the object is valid on function entry and exit. I would leave that as an exercise and a separate PR for later, if you'd like to do that.

Otherwise if we try to handle all the possible value types going sour in place where clear logic should be, we'll get entangled in condition spaghetti.